### ISSUE TITLE
webhook IDs can't be paths?

### ISSUE THREAD

**Issue by enjoys-sashimi** [adnanh/webhook#421]
I'm confused why I can't get webhook id's with paths in them to work?

```
[webhook] 2020/05/19 01:46:46 found 3 hook(s) in file
[webhook] 2020/05/19 01:46:46   loaded: test
[webhook] 2020/05/19 01:46:46   loaded: test_2
[webhook] 2020/05/19 01:46:46   loaded: test_2/canwepath
```

You can see all three hooks are found; however, the only ones that I can actually `POST` to are `test` and `test_2`. If I try to send a request to `test_2/canwepath` I get  `404 Not Found`

Would appreciate any help here; not sure what I'm missing. Kind of a bummer because the API I'm trying to receive requests from uses paths in their IDs

**Comment by moorereason** [comment]
I don't think this functionality has changed recently, but what version of `webhook` are you using?

We assume the hook ID will be a single path segment ([see the code here](https://github.com/adnanh/webhook/blob/345bf3d4094de98df6bd56b91e04b4c0cfe6cc62/webhook.go#L780-L785)).  So, you can build hooks with path separators, but the HTTP server mux currently only supports a single path segment.   We should at least warn about path separators in the hook IDs unless we change how the mux behaves.

**Comment by moorereason** [comment]
To accommodate this request, it looks like we'd need to use [`mux.PathPrefix`](https://godoc.org/github.com/gorilla/mux#Router.PathPrefix) and change how we parse the ID out of the path.

**Comment by king-salmon** [comment]
@moorereason Not sure; I didn't pull from source but I grabbed the latest community packaged version w/ `apt`.
So I'm not sure I understand, adding a path separator (like `-` ) in my IDs will provide me with a path? 

So in hooks.json/webhooks.conf I could change
```
         "id": "test_2/canwepath",
```
to
```
         "id": "test_2-canwepath",
```

**Comment by moorereason** [comment]
**TL;DR**: If you include the `/` path separator in your hook ID, it will not work today.  There are no work-arounds.

**More**

Sorry, I wasn't clear.  I was using some technical jargon about URI paths that probably needed explanation.

When I say "path separator," I mean `/` in this case (If we were talking about a Windows filesystem, I would mean `\`).  By "path segment" I generally mean the segments between path separators.

So, in a request for `/hooks/test_2/canwepath`, the segments would be `hooks`, `test_2`, and `canwepath`.  The `hooks` prefix is an optional path segment in `webhook`, and `webhook` only allows a single path segment for the hook ID after the prefix.

If you include the `/` path separator in your hook ID, it will not work today.  There are no work-arounds.

We would need to rework how `webhook` handles the request URI in order to support your `test_2/canwepath` case.

**Comment by king-salmon** [comment]
Ahh that makes sense, thank you for breaking that down for me.
